### PR TITLE
Move code contribution guide into repo

### PR DESCRIPTION
**Pull request by damccorm**
This should hopefully make things more discoverable

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by damccorm** [comment]
R: @liferoad

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Thank you for your contribution! Follow this checklist to help us incorporate yo
  - [ ] Update `CHANGES.md` with noteworthy changes.
  - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).
 
-See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).
+See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.orghttps://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).
```
Comment (liferoad) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: is this supposed to be `https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier)`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -17,13 +17,314 @@
...
+These steps and instructions on getting started are outlined below as well.
+
+### Prerequisites
+
+- A [GitHub](https://github.com/) account.
+- A Linux, macOS, or Microsoft Windows development environment.
+- Java JDK 8 installed.
+- [Docker](https://www.docker.com/) installed for some tasks including building worker containers and testing changes to this website locally.
+- For SDK Development:
+  - [Go](https://golang.org) 1.16.0 or later installed for Go SDK development.
```
Comment (liferoad) [inline CONTRIBUTING.md:46]: I remember Go has to be installed even for Java SDK development.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -17,13 +17,314 @@
...
+#### Get Reviewed
+
+Your pull requests should automatically have reviewers assigned within a few hours of opening it.
+If that doesn't happen for some reason, you can also request a review yourself.
+
+1. Pull requests can only be merged by a
+   [Beam committer](https://home.apache.org/phonebook.html?pmc=beam).
+   To find a committer for your area, either:
+  - look for similar code merges, or
+  - ask on [dev@beam.apache.org](/community/contact-us/)
```
Comment (liferoad) [inline CONTRIBUTING.md:263]: should this be `https://beam.apache.org/community/contact-us/`?

Comment (liferoad) [reply CONTRIBUTING.md:263]: looks like other links might be broken.

**Review by liferoad — APPROVED** [review]
Thanks for doing this! Left some minor comments.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (damccorm) [reply .github/PULL_REQUEST_TEMPLATE.md:12]: Fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (damccorm) [reply CONTRIBUTING.md]: Moved it out of SDK development and removed `for Go SDK development`

File: CONTRIBUTING.md (hunk 3 above)
Comment (damccorm) [reply CONTRIBUTING.md]: Good call, I didn't notice these when I migrated some content from the website. Thanks!

**Pull request merged**